### ISSUE TITLE
Supports connection option 'CurrentSchema'

### ISSUE THREAD

**Issue by shawnzhu** [ibmdb/python-ibmdbsa#76]
The connection configuration [`CurrentSchema`](https://www.ibm.com/support/knowledgecenter/en/SSEPGG_10.5.0/com.ibm.db2.luw.apdv.cli.doc/doc/r0008779.html) is a supported configuration for pip package `ibm-db` but not in this package where it supports only.

E.g., when using `ibm-db`:

```Python
import ibm_db
import ibm_db_dbi
import os

conn_str = 'DATABASE={};HOSTNAME={};PORT={};Security=SSL;UID={};PWD={};CurrentSchema={}' \
    .format(
        'BLUDB',
        'db_host',
        50001,
        'bluadmin',
        os.environ['DB2_PASSWORD'],
        'NOT_BLUADMIN',
    )

conn_handler = ibm_db.connect(conn_str, '', '')
conn = ibm_db_dbi.Connection(conn_handler)

sql_stmt = '''
SELECT * FROM ACTOR LIMIT 10
'''

cur = conn.cursor()
cur.execute(sql_stmt)
row=cur.fetchall()
print(row)
```

it works well by querying table ``

## Use case
given table `ACTOR` in the schema `NOT_BLUADMIN`
when using this package:

```Python
import os

from sqlalchemy import *
from sqlalchemy.engine import create_engine
from sqlalchemy.schema import *

# Db2
e = create_engine(
    'db2+ibm_db://bluadmin:xxxx@db_host:50001/BLUDB?Security=SSL&CurrentSchema=NOT_BLUADMIN'
)
sql = text("select * from ACTOR LIMIT 10")
result = e.execute(sql)
print([row for row in result])
```
## Expected result

it shall query table `NOT_BLUADMIN.ACTOR` successfully.

## Actual result
It will report error which shows it still use current user name as current schema:

```
sqlalchemy.exc.ProgrammingError: (ibm_db_dbi.ProgrammingError) ibm_db_dbi::ProgrammingError: Exception('SQLNumResultCols failed: [IBM][CLI Driver][DB2/LINUXX8664] SQL0204N  "BLUADMIN.ACTOR" is an undefined name.  SQLSTATE=42704 SQLCODE=-204')
[SQL: select * from ACTOR LIMIT 10]
```

**Comment by imavo** [comment]
The ibm_db_sa module already supports the attribute CURRENTSCHEMA.
Your code example will not work however because you specify it incorrectly in the connection string.  

The following connection string works for me, it shows the attribute follows the database name and has separator semicolon both before and after the attribute-setting.  After connection the schema change is effective for me, so ibm_db_sa addresses the target schema instead of the implicit schema of the authid.

`db2 = sqlalchemy.create_engine('ibm_db_sa://'+ db_user + ':' + db_pwd + '@localhost:50100/' + db_name +";CURRENTSCHEMA=OTHERUSER;" )`

**Comment by shawnzhu** [comment]
@imavo Thanks for the feedback, we're talking different database URLs.

the db URL `ibm_db_sa://xxx` is for SQLAlchemy 0.7/0.8 see https://github.com/ibmdb/python-ibmdbsa/commit/1c4ccd9e1634e01440773ce049b5dfef556fa504 the connection string works for ibm_db when using the clidriver. I don't mind using this legacy URL but can not find it anywhere from this repo. do you mind adding a PR to document it?

> Your code example will not work however because you specify it incorrectly in the connection string.

This PR is for the database URL that follows RFC 1738 which is documented and used by most people. See https://docs.sqlalchemy.org/en/13/core/engines.html#database-urls where Db2 connection string will be implementation details and all users will follow the SQLAlchemy doc to connect to Db2.

@imavo do you mind reviewing this PR?

**Comment by imavo** [comment]
Both styles work for me:

`db2 = sqlalchemy.create_engine('db2+ibm_db://'+ db_user + ':' + db_pwd + '@localhost:50100/' + db_name +";CURRENTSCHEMA=OTHERUSER;" )`

and the legacy style:

`db2 = sqlalchemy.create_engine('ibm_db_sa://'+ db_user + ':' + db_pwd + '@localhost:50100/' + db_name +";CURRENTSCHEMA=OTHERUSER;" )`

With a Db2-Linux/Unix/Windows v11.5 clidriver , the connection attributes are documented  [here](https://www.ibm.com/support/knowledgecenter/SSEPGG_11.5.0/com.ibm.db2.luw.apdv.cli.doc/doc/r0007964.html)

You can append additional connection attributes as long as the semicolon prefix and suffix exists.

**Comment by shawnzhu** [comment]
> You can append additional connection attributes as long as the semicolon prefix and suffix exists.

My point is supporting the [SQLAlchemy Engine Creation API](https://docs.sqlalchemy.org/en/13/core/engines.html#engine-creation-api) where it uses query syntax of RFC3986 to config parameters (what SQLAlchemy users follow), not to to tell SQLAlchemy users to follow driver specific config syntax like semicolon prefix and suffix. 

Again, I still hope this PR can be merged which will benefit SQLAlchemy users like me. it would be great if you can create another PR to add ☝️connection attributes as document (and I will be happy to review it)

**Comment by shawnzhu** [comment]
@imavo Thanks. I tried ☝️ dialect specific parameter config syntax and it did work. Just hope it can be documented.

> Your code example will not work however because you specify it incorrectly in the connection string.

my code example will work if this PR is merged. My point is using SQLAlchemy engine API to configure parameters instead of using dialect specific config.

(I was a little big disappointed that being told a general syntax to configure connection string in SQLAlchemy is incorrect while a dialect specific config syntax is the ONLY supported option, which defeat the purpose of SQLAlchemy, IMHO).

**Comment by imavo** [comment]
Apologies for saying your approach was incorrect, that was wrong of me. Many people just want to know how to make CURRENTSCHEMA connection attribute work without waiting for code changes. This may help them. The URL syntax I showed has been in Db2 for decades for both CLI/ODBC and jdbc interfaces and other interfaces, so will be familiar to any Db2 programmer, and is documented by IBM in the various Db2 Knowledge Centers, so I see no value in duplicating it.

About your PR, I will leave that to the code maintainers to assess.

**Comment by shawnzhu** [comment]
@imavo Db2 centric approach will definitely do the very job like what your code sample can do. I just simply follow a different approach that being dialect neutral.

**Comment by shawnzhu** [comment]
This has been fixed in release v0.3.6 by https://github.com/ibmdb/python-ibmdbsa/commit/3362a2efd7d2efb15bc0dbfe42887fca672412ee
